dependency_cache.go

```diff
@@ -181,7 +202,7 @@ func filterBindingsByType(bindings libcnb.Bindings, bindingType string) (map[str
 				if _, ok := filteredBindings[key]; ok {
 					return nil, fmt.Errorf("multiple %s bindings found with duplicate keys %s", binding.Type, key)
 				}
-				filteredBindings[key] = value
+				filteredBindings[strings.ToLower(key)] = value
```

Comment:
Two things about this:

1. Small thing. I think we'd need to `strings.ToLower(key)` the check on line 202 as well. Otherwise, it would use a potentially non-lowercase key to look in the map which we're then only putting lowercase keys into.

2. I worry about changing the behavior here. Previously, this was filtering case sensitively. We could impact users with existing bindings. Is there a strong need to change how this works? I think my preference would be to not change this here, unless there's a strong reason to. If nothing else, for backward compatibility.

Reply:
Yes, you are right about line 202; this needs to be added.
The switch to case insensitivity is caused by the requirement to override dependency-mirror bindings with those set in env variables (line 163). If the map was initially filled with case sensitive keys (like before), we would have to iterate over those again afterwards instead of accessing the map entry directly.

Regarding the backwards compatibility:
I believe, this code was used only on bindings of type "dependency-mapping" so far, which work with SHA256 checksums as keys. Since these are only hex representations, they can't be considered case sensitive. The only problem I can think of is, if someone didn't put the SHA256 in lowercase in the buildpacks.toml. But this would already fail at verification in line 310.

Reply:
> Regarding the backwards compatibility:

That's a good point and you're right. The tooling we use should ensure that all the hashes are in the same case too. I'm OK with changing this.